src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/update/action/ValidateYFlowAction.java

```diff
@@ -0,0 +1,150 @@
+        featureTogglesRepository = repositoryFactory.createFeatureTogglesRepository();
+        yFlowRepository = repositoryFactory.createYFlowRepository();
+        yFlowValidator = new YFlowValidator(persistenceManager);
+        this.dashboardLogger = dashboardLogger;
+    }
+
+    @Override
+    protected Optional<Message> performWithResponse(State from, State to, Event event, YFlowUpdateContext context,
+                                                    YFlowUpdateFsm stateMachine) {
+        YFlowRequest targetFlow = context.getTargetFlow();
+
+        boolean isOperationAllowed = featureTogglesRepository.getOrDefault().getModifyYFlowEnabled();
+        if (!isOperationAllowed) {
+            throw new FlowProcessingException(ErrorType.NOT_PERMITTED, "Y-flow create feature is disabled");
+        }
+
+        try {
+            yFlowValidator.validate(targetFlow);
+        } catch (InvalidFlowException e) {
+            throw new FlowProcessingException(e.getType(), e.getMessage(), e);
+        } catch (UnavailableFlowEndpointException e) {
+            throw new FlowProcessingException(ErrorType.DATA_INVALID, e.getMessage(), e);
+        }
+
+        String yFlowId = targetFlow.getYFlowId();
+        YFlow yFlow = transactionManager.doInTransaction(() -> {
+            YFlow result = yFlowRepository.findById(yFlowId)
+                    .orElseThrow(() -> new FlowProcessingException(ErrorType.NOT_FOUND,
+                            format("Y-flow %s not found", yFlowId)));
+            if (result.getStatus() == FlowStatus.IN_PROGRESS) {
+                throw new FlowProcessingException(ErrorType.REQUEST_INVALID,
+                        format("Y-flow %s is in progress now", yFlowId));
+            }
+
+            // Keep it, just in case we have to revert it.
+            stateMachine.setOriginalYFlowStatus(result.getStatus());
+
+            result.setStatus(FlowStatus.IN_PROGRESS);
+            return result;
+        });
+
+        Set<String> requestedSubFlowIds = targetFlow.getSubFlows().stream()
+                .map(SubFlowDto::getFlowId)
+                .collect(Collectors.toSet());
+
+        Set<String> originalSubFlowIds = yFlow.getSubFlows().stream()
+                .map(YSubFlow::getSubFlowId)
+                .collect(Collectors.toSet());
+
+        if (!requestedSubFlowIds.equals(originalSubFlowIds)) {
+            throw new FlowProcessingException(ErrorType.PARAMETERS_INVALID,
+                    format("Unable to map provided sub-flows set onto existing y-flow %s", yFlowId));
+        }
+
+        List<FlowEndpoint> subFlowEndpoints = targetFlow.getSubFlows().stream()
+                .map(SubFlowDto::getEndpoint)
+                .collect(Collectors.toList());
+        dashboardLogger.onYFlowUpdate(yFlowId, targetFlow.getSharedEndpoint(), subFlowEndpoints,
+                targetFlow.getMaximumBandwidth());
+
+        stateMachine.setTargetFlow(targetFlow);
```

Comment:
Move "old" resource fill up in separate method. It do not "fit" into "validation" process at all.

Reply:
Moved to UpdateAction, as done in the update operation for flow.